### PR TITLE
Community info page

### PR DESCRIPTION
**Pull request by darrenvong**
Half of #927. Community edit page to come in the next PR as this is getting bigger than expected (although a good 1/3 of the changes are tests/Storybook code).

**Desktop**
Screenshot 2021-06-05 at 04 04 17

**Mobile**
Screenshot 2021-06-05 at 04 03 53

**Frontend checklist**
- [x] Formatted my code with `yarn format && yarn lint --fix`
- [x] There are no warnings from `yarn lint`
- [x] There are no console warnings when running the app
- [x] Added any new components to storybook
- [x] Added tests where relevant
- [x] All tests pass
- [x] Clicked around my changes running locally and it works
- [x] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

File: app/frontend/src/features/communities/CommunityInfoPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    beforeEach(async () => {
+      listAdminsMock.mockResolvedValue({
+        adminUserIdsList: [2, 3],
+        nextPageToken: "3",
+      });
+      render(<CommunityInfoPage community={community} />, { wrapper });
+      await waitForElementToBeRemoved(screen.getByRole("progressbar"));
+
+      userEvent.click(screen.getByRole("button", { name: SEE_ALL_MODERATORS }));
+      await screen.findByRole("presentation");
```
Comment (lucaslcode) [inline app/frontend/src/features/communities/CommunityInfoPage.test.tsx]: Does this (and similar) need to be more specific? Match the title or something?

File: app/frontend/src/features/communities/CommunityInfoPage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      // Close dialog by clicking the backdrop
+      userEvent.click(document.querySelector(".MuiBackdrop-root")!);
+      await waitForElementToBeRemoved(screen.getByRole("presentation"));
+
+      expect(
+        screen.queryByRole("button", { name: LOAD_MORE_MODERATORS })
+      ).not.toBeInTheDocument();
+    });
+
+    it("closes the dialog by pressing the escape key", async () => {
```
Comment (lucaslcode) [inline app/frontend/src/features/communities/CommunityInfoPage.test.tsx:169]: Isn't this testing mui?

File: app/frontend/src/features/communities/CommunityInfoPage.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  title: {
+    alignItems: "center",
+    display: "flex",
+    "& > * + *": {
+      marginInlineStart: theme.spacing(1),
+    },
+  },
+  content: {
+    "& p": {
+      margin: 0,
```
Comment (lucaslcode) [inline app/frontend/src/features/communities/CommunityInfoPage.tsx]: Why is this needed here and not the markdown component in general?

File: app/frontend/src/features/communities/CommunityLeadersSection.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        <SectionTitle icon={<CommunityLeadersIcon />}>
+          {COMMUNITY_LEADERS}
+        </SectionTitle>
+        <Typography variant="body1">{COMMUNITY_LEADERS_DESCRIPTION}</Typography>
+        <Typography variant="h3">{COMMUNITY_MODERATORS}</Typography>
+        {error ? (
+          <Alert severity="error">{error.message}</Alert>
+        ) : isLoading ? (
+          <CircularProgress />
+        ) : adminIds && adminIds.length > 0 ? (
```
Comment (lucaslcode) [inline app/frontend/src/features/communities/CommunityLeadersSection.tsx]: Time to stop destructuring react-query? Or you prefer this way?

File: app/frontend/src/features/communities/CommunityLeadersSection.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+}: CommunityLeadersSectionProps) {
+  const classes = useStyles();
+  const { adminIds, adminUsers, error, isLoading, hasNextPage } = useListAdmins(
+    community.communityId,
+    "summary"
+  );
+  const [isModeratorsDialogOpen, setIsModeratorsDialogOpen] = useState(false);
+
+  return (
+    <>
```
Comment (lucaslcode) [inline app/frontend/src/features/communities/CommunityLeadersSection.tsx]: Unneeded fragment

File: app/frontend/src/features/communities/CommunityPage/EventsSection.tsx [hunk 6]
```diff
@@ -15,7 +15,7 @@ import { useCommunityPageStyles } from "./CommunityPage";
 import EventCard from "./EventCard";
 import SectionTitle from "./SectionTitle";
 
-export default function PlacesSection({
+export default function EventsSection({
```
Comment (lucaslcode) [inline app/frontend/src/features/communities/CommunityPage/EventsSection.tsx:18]: oops

File: app/frontend/src/features/communities/constants.ts [hunk 7]
```diff
@@ -2,6 +2,7 @@ import { CommunityTab } from "routes";
 
 export const communityTabBarLabels: Record<CommunityTab, string> = {
   overview: "Overview",
+  info: "General",
```
Comment (lucaslcode) [inline app/frontend/src/features/communities/constants.ts]:
```suggestion
  info: "Local info",
```

File: app/frontend/src/features/communities/constants.ts [hunk 8]
```diff
@@ -14,6 +15,10 @@ export const CLOSE = "Close";
 export const COMMENT = "Comment";
 export const COMMENTS = "Comments";
 export const COMMUNITY_HEADING = (name: string) => `Welcome to ${name}!`;
+export const COMMUNITY_LEADERS = "Community leaders";
+export const COMMUNITY_LEADERS_DESCRIPTION =
+  "Below is a list of current leaders within this community.";
```
Comment (lucaslcode) [inline app/frontend/src/features/communities/constants.ts]: Is this from the desktop mockup? "Leaders" sounds pretty weird to me...

**Review by lucaslcode — CHANGES_REQUESTED** [review]

File: app/frontend/src/features/communities/CommunityInfoPage.test.tsx (hunk 1 above)
Comment (darrenvong) [reply app/frontend/src/features/communities/CommunityInfoPage.test.tsx]: I tried and it didn't work, ends up with a cannot find element with `role="presentation"` :( Might be because the dialog itself also a role "none" on it that interferes with the *ByRole stuff...

File: app/frontend/src/features/communities/CommunityInfoPage.test.tsx (hunk 2 above)
Comment (darrenvong) [reply app/frontend/src/features/communities/CommunityInfoPage.test.tsx:169]: I thought about this, but in some places like the bug report dialog we explicitly disable some things like no clicking background for closing, so I guess this is checking this hasn't been disabled by accident

File: app/frontend/src/features/communities/CommunityInfoPage.tsx (hunk 3 above)
Comment (darrenvong) [reply app/frontend/src/features/communities/CommunityInfoPage.tsx]: I haven't checked if this will break in other places and if we want to disable the margin too, my guess is probably yes? I'll move it into `Markdown` if that's the case 👍

File: app/frontend/src/features/communities/CommunityLeadersSection.tsx (hunk 4 above)
Comment (darrenvong) [reply app/frontend/src/features/communities/CommunityLeadersSection.tsx]: I modified the `useListAdmins` hook slightly so in this case it's returning more than the `listAdmins` query object, and since `adminIds` isn't from the main query object, it doesn't make a difference in this case.

But yeah, definitely worth trying out the type narrowing trick we read on the other post more for these kind of pattern more for a bog standard query object!

File: app/frontend/src/features/communities/constants.ts (hunk 8 above)
Comment (darrenvong) [reply app/frontend/src/features/communities/constants.ts]: Yeah, and then it has another heading on "Community moderators" right underneath that. Maybe I should move "Community moderators" into the h2 title and remove this description? It's kinda obvious from looking at the UI with what the title says right?

Comment (lucaslcode) [reply app/frontend/src/features/communities/constants.ts]: Yes definitely

File: app/frontend/src/features/communities/CommunityPage/InfoPageSection.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+      <SectionTitle icon={<InfoIcon />}>{GENERAL_INFORMATION}</SectionTitle>
+      <Typography className={classes.summaryText} variant="body1">
+        {getContentSummary(community.mainPage?.content)}
+      </Typography>
+
+      <div className={classes.loadMoreButton}>
+        <Link
+          to={routeToCommunity(community.communityId, community.slug, "info")}
+        >
+          <Button component="span">{SEE_MORE_INFORMATION}</Button>
```
Comment (lucaslcode) [inline app/frontend/src/features/communities/CommunityPage/InfoPageSection.tsx]: Do you think this makes more sense as an MuiLink? (And I guess also the other "See more" buttons)

Comment (darrenvong) [reply app/frontend/src/features/communities/CommunityPage/InfoPageSection.tsx]: Hmm yeah, since it's changing the user's URL (regardless of with or without React Router) so a link makes more sense to me. Maybe I'll start changing it here...

File: app/frontend/src/features/communities/CommunityInfoPage.test.tsx (hunk 1 above)
Comment (darrenvong) [reply app/frontend/src/features/communities/CommunityInfoPage.test.tsx]: Never mind, doing it on role `dialog` works instead, so yes better to match with the title so we know it's the right one 👍

File: app/frontend/src/components/Markdown.tsx [hunk 10]
```diff
@@ -27,7 +27,7 @@ const useStyles = makeStyles((theme) => ({
     "& h4": theme.typography.h4,
     "& h5": theme.typography.h5,
     "& h6": theme.typography.h6,
-    "& p": theme.typography.body1,
+    "& p": { ...theme.typography.body1, margin: 0 },
```
Comment (lucaslcode) [inline app/frontend/src/components/Markdown.tsx]: On second thought, it's better with the paragraph margin. Why did you want to get rid of it in the local info?

**Review by lucaslcode — CHANGES_REQUESTED** [review]
Looks good, one tiny thing

File: app/frontend/src/components/Markdown.tsx (hunk 10 above)
Comment (darrenvong) [reply app/frontend/src/components/Markdown.tsx]: I thought it made the gap between the paragraph and the title a little too big, and the fact it's closer to the moderators section makes it looks like it belongs more to that section rather than the "general information" heading immediately below.

**With margin**
Screenshot 2021-06-06 at 16 18 16

**Without margin (current code)**
Screenshot 2021-06-06 at 16 18 23

Comment (darrenvong) [reply app/frontend/src/components/Markdown.tsx]: ~~Maybe a better solution is to push the "community moderators" further down?~~ Never mind, pushing the section down doesn't really help after more glancing at it, I think the thing that bothers me is the lack of symmetry between the distance of the paragraph from the page header compared to the Mui parapgraph under "community moderators"

The problem is the page title already has some padding, so that's why having the paragraph also with the margin makes the gap look massive when compared to the normal "body1" paragraph under "community moderators", so actually I think the specific component override is fine? I just checked again and removing the margin makes the markdown paragraph a bit too close to the back button on a discussion page description now...

**Comment by darrenvong** [comment]
By the way if you want to get this in soon, feel free to hit merge if you are happy as I might not get around doing it today

**Review by lucaslcode — CHANGES_REQUESTED** [review]
Failing test

File: app/frontend/src/components/Markdown.tsx (hunk 10 above)
Comment (lucaslcode) [reply app/frontend/src/components/Markdown.tsx]: I don't mind either way for the headings, but I'm thinking about when you have two paragraphs together, there should be some spacing. If it's set to 0, then it's equivalent to `<br>` and it shouldn't be

Comment (lucaslcode) [reply app/frontend/src/components/Markdown.tsx]: @darrenvong ☝️

**Review by lucaslcode — CHANGES_REQUESTED** [review]
Merge conflicts, and did you see my comment about the `<p>` margin?

**Comment by darrenvong** [comment]
> did you see my comment about the `<p>` margin?

Yes, I ended up adding `& p` selector so it shares the same `marginTop` that we get in other heading component as well so should fix the issue of two paragraphs being right below each other you mentioned? I didn't need to hack around the margin stuff as much in the end by using `SectionTitle` and making it more flexible to accept different heading levels so it's still semantically correct depending on where it's used.

**Review by lucaslcode — APPROVED** [review]
